### PR TITLE
[WIP] Adding Search app.

### PR DESCRIPTION
**Pull request by kegor**
**Problem:**
We need to implement back-end code of the new Search sample app.

**Solution:**
Needed code was implemented as the MVC project using the Search library API.

**Screenshots:**
<details>
  <summary>Take a look</summary>

_Search by phrase:_
search_by_phrase

_Search by file name:_
search_by_file_name

_Search by file name with extension:_
search_by_file_name_with_ext

_Searching by phrase with special char and clearing the search query:_
searching_phrases_clearing_search_query

_Adding password-protected files into index:_
adding_password_protected_file_to_index

</details>

**Related PR:**
Additional screenshots and needed UI code you could find in the related PR: https://github.com/groupdocs-total/GroupDocs.Total-Angular/pull/162.

### CODE REVIEW

**Comment by egorovpavel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/Products/Search/Entity/Web/Request/SearchPostedData.cs  1
- src/Products/Search/Entity/Web/IndexedFileDescriptionEntity.cs  1
- src/Products/Search/Service/SearchService.cs  5
- src/Products/Search/Controllers/SearchApiController.cs  7
- src/Products/Search/Util/Directory/DirectoryUtils.cs  4
- src/Products/Search/Entity/Web/Response/SearchDocumentResult.cs  1
- src/Products/Search/Config/SearchConfiguration.cs  3
- src/Products/Search/Entity/Web/Response/SummarySearchResult.cs  1
         

Clones added
============
- src/Products/Search/Controllers/SearchApiController.cs  7
- src/Products/Search/Util/Directory/DirectoryUtils.cs  1
- src/Products/Search/Config/SearchConfiguration.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GroupDocs/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5558558&bid=18230952)

**Comment by bobkovalex** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/Products/Search/Util/Directory/DirectoryUtils.cs  4
- src/Products/Search/Controllers/SearchApiController.cs  7
- src/Products/Search/Entity/Web/IndexedFileDescriptionEntity.cs  1
- src/Products/Search/Entity/Web/Request/SearchPostedData.cs  1
- src/Products/Search/Service/SearchService.cs  5
- src/Products/Search/Entity/Web/Response/SearchDocumentResult.cs  1
- src/Products/Search/Config/SearchConfiguration.cs  3
- src/Products/Search/Entity/Web/Response/SummarySearchResult.cs  1
         

Clones added
============
- src/Products/Search/Util/Directory/DirectoryUtils.cs  1
- src/Products/Search/Controllers/SearchApiController.cs  7
- src/Products/Search/Config/SearchConfiguration.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5558557&bid=18230951)

File: src/Products/Search/Service/SearchService.cs [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+using GroupDocs.Total.MVC.Products.Common.Entity.Web;
+using GroupDocs.Total.MVC.Products.Common.Config;
+using System;
+
+namespace GroupDocs.Total.MVC.Products.Search.Service
+{
+    public static class SearchService
+    {
+        private static Index index;
+        internal static Dictionary<string, DocumentStatus> filesIndexStatuses = new Dictionary<string, DocumentStatus>();
```
Comment (egorovpavel) [inline src/Products/Search/Service/SearchService.cs:19]: Codacy Issue found: [Change the visibility of 'filesIndexStatuses' or make it 'const' or 'readonly'.](https://app.codacy.com/manual/GroupDocs/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5558558)

**Comment by egorovpavel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- src/Products/Search/Controllers/SearchApiController.cs  8
- src/Products/Search/Entity/Web/IndexedFileDescriptionEntity.cs  1
- src/Products/Search/Util/Directory/DirectoryUtils.cs  4
- src/Products/Search/Config/SearchConfiguration.cs  5
- src/Products/Search/Service/SearchService.cs  7
- src/Products/Search/Entity/Web/Response/SummarySearchResult.cs  1
- src/Products/Search/Entity/Web/Request/SearchPostedData.cs  1
- src/Products/Search/Entity/Web/Response/SearchDocumentResult.cs  1
         

Clones added
============
- src/Products/Search/Controllers/SearchApiController.cs  7
- src/Products/Search/Util/Directory/DirectoryUtils.cs  1
- src/Products/Search/Config/SearchConfiguration.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GroupDocs/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5558558&bid=18230952)

File: src/Products/Search/Service/SearchService.cs [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+        public static SummarySearchResult Search(SearchPostedData searchRequest, Common.Config.GlobalConfiguration globalConfiguration)
+        {
+            if (index == null) return new SummarySearchResult();
```
Comment (egorovpavel) [inline src/Products/Search/Service/SearchService.cs:23]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/GroupDocs/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5558558)

**Comment by bobkovalex** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- src/Products/Search/Entity/Web/Response/SummarySearchResult.cs  1
- src/Products/Search/Entity/Web/Response/SearchDocumentResult.cs  1
- src/Products/Search/Controllers/SearchApiController.cs  8
- src/Products/Search/Entity/Web/Request/SearchPostedData.cs  1
- src/Products/Search/Service/SearchService.cs  7
- src/Products/Search/Util/Directory/DirectoryUtils.cs  4
- src/Products/Search/Config/SearchConfiguration.cs  5
- src/Products/Search/Entity/Web/IndexedFileDescriptionEntity.cs  1
         

Clones added
============
- src/Products/Search/Controllers/SearchApiController.cs  7
- src/Products/Search/Util/Directory/DirectoryUtils.cs  1
- src/Products/Search/Config/SearchConfiguration.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5558557&bid=18230951)

File: src/Products/Search/Service/SearchService.cs (hunk 2 above)
Comment (bobkovalex) [inline src/Products/Search/Service/SearchService.cs:23]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5558557)

File: src/Products/Search/Service/SearchService.cs (hunk 1 above)
Comment (bobkovalex) [inline src/Products/Search/Service/SearchService.cs:19]: Codacy Issue found: [Change the visibility of 'filesIndexStatuses' or make it 'const' or 'readonly'.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5558557)

**Comment by egorovpavel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/Products/Search/Entity/Web/Response/SummarySearchResult.cs  1
- src/Products/Search/Util/Directory/DirectoryUtils.cs  4
- src/Products/Search/Entity/Web/Response/SearchDocumentResult.cs  1
- src/Products/Search/Entity/Web/Request/SearchPostedData.cs  1
- src/Products/Search/Config/SearchConfiguration.cs  5
- src/Products/Search/Entity/Web/IndexedFileDescriptionEntity.cs  1
- src/Products/Search/Service/SearchService.cs  7
- src/Products/Search/Controllers/SearchApiController.cs  8
         

Clones added
============
- src/Products/Search/Util/Directory/DirectoryUtils.cs  1
- src/Products/Search/Config/SearchConfiguration.cs  1
- src/Products/Search/Controllers/SearchApiController.cs  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GroupDocs/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5558558&bid=18230952)

**Comment by bobkovalex** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/Products/Search/Service/SearchService.cs  7
- src/Products/Search/Controllers/SearchApiController.cs  8
- src/Products/Search/Entity/Web/Request/SearchPostedData.cs  1
- src/Products/Search/Util/Directory/DirectoryUtils.cs  4
- src/Products/Search/Entity/Web/IndexedFileDescriptionEntity.cs  1
- src/Products/Search/Entity/Web/Response/SearchDocumentResult.cs  1
- src/Products/Search/Entity/Web/Response/SummarySearchResult.cs  1
- src/Products/Search/Config/SearchConfiguration.cs  5
         

Clones added
============
- src/Products/Search/Controllers/SearchApiController.cs  7
- src/Products/Search/Util/Directory/DirectoryUtils.cs  1
- src/Products/Search/Config/SearchConfiguration.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5558557&bid=18230951)

**Comment by bobkovalex** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/Products/Search/Controllers/SearchApiController.cs  10
- src/Products/Search/Entity/Web/Request/SearchPostedData.cs  1
- src/Products/Search/Entity/Web/Response/SearchDocumentResult.cs  1
- src/Products/Search/Config/SearchConfiguration.cs  7
- src/Products/Search/Entity/Web/IndexedFileDescriptionEntity.cs  1
- src/Products/Search/Util/Directory/DirectoryUtils.cs  4
- src/Products/Search/Entity/Web/Response/SummarySearchResult.cs  1
- src/Products/Search/Service/SearchService.cs  7
         

Clones added
============
- src/Products/Search/Controllers/SearchApiController.cs  7
- src/Products/Search/Config/SearchConfiguration.cs  1
- src/Products/Search/Util/Directory/DirectoryUtils.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5558557&bid=18230951)

**Comment by egorovpavel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/Products/Search/Service/SearchService.cs  7
- src/Products/Search/Entity/Web/IndexedFileDescriptionEntity.cs  1
- src/Products/Search/Config/SearchConfiguration.cs  7
- src/Products/Search/Entity/Web/Request/SearchPostedData.cs  1
- src/Products/Search/Controllers/SearchApiController.cs  10
- src/Products/Search/Entity/Web/Response/SearchDocumentResult.cs  1
- src/Products/Search/Util/Directory/DirectoryUtils.cs  4
- src/Products/Search/Entity/Web/Response/SummarySearchResult.cs  1
         

Clones added
============
- src/Products/Search/Config/SearchConfiguration.cs  1
- src/Products/Search/Controllers/SearchApiController.cs  7
- src/Products/Search/Util/Directory/DirectoryUtils.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GroupDocs/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5558558&bid=18230952)

**Comment by bobkovalex** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/Products/Search/Entity/Web/Request/SearchPostedData.cs  1
- src/Products/Search/Config/SearchConfiguration.cs  7
- src/Products/Search/Service/SearchService.cs  7
- src/Products/Search/Entity/Web/Response/SummarySearchResult.cs  1
- src/Products/Search/Controllers/SearchApiController.cs  10
- src/Products/Search/Entity/Web/IndexedFileDescriptionEntity.cs  1
- src/Products/Search/Util/Directory/DirectoryUtils.cs  4
- src/Products/Search/Entity/Web/Response/SearchDocumentResult.cs  1
         

Clones added
============
- src/Products/Search/Config/SearchConfiguration.cs  1
- src/Products/Search/Controllers/SearchApiController.cs  7
- src/Products/Search/Util/Directory/DirectoryUtils.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5558557&bid=18230951)

**Comment by egorovpavel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/Products/Search/Config/SearchConfiguration.cs  7
- src/Products/Search/Util/Directory/DirectoryUtils.cs  4
- src/Products/Search/Entity/Web/IndexedFileDescriptionEntity.cs  1
- src/Products/Search/Entity/Web/Response/SummarySearchResult.cs  1
- src/Products/Search/Service/SearchService.cs  7
- src/Products/Search/Entity/Web/Response/SearchDocumentResult.cs  1
- src/Products/Search/Entity/Web/Request/SearchPostedData.cs  1
- src/Products/Search/Controllers/SearchApiController.cs  10
         

Clones added
============
- src/Products/Search/Config/SearchConfiguration.cs  1
- src/Products/Search/Util/Directory/DirectoryUtils.cs  1
- src/Products/Search/Controllers/SearchApiController.cs  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GroupDocs/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5558558&bid=18230952)

**Comment by bobkovalex** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/Products/Search/Entity/Web/IndexedFileDescriptionEntity.cs  1
- src/Products/Search/Controllers/SearchApiController.cs  10
- src/Products/Search/Service/SearchService.cs  7
- src/Products/Search/Config/SearchConfiguration.cs  7
- src/Products/Search/Entity/Web/Response/SummarySearchResult.cs  1
- src/Products/Search/Util/Directory/DirectoryUtils.cs  4
- src/Products/Search/Entity/Web/Request/SearchPostedData.cs  1
- src/Products/Search/Entity/Web/Response/SearchDocumentResult.cs  1
         

Clones added
============
- src/Products/Search/Controllers/SearchApiController.cs  7
- src/Products/Search/Config/SearchConfiguration.cs  1
- src/Products/Search/Util/Directory/DirectoryUtils.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5558557&bid=18230951)

**Comment by bobkovalex** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- src/Products/Search/Entity/Web/IndexedFileDescriptionEntity.cs  1
- src/Products/Search/Entity/Web/Request/SearchPostedData.cs  1
- src/Products/Search/Config/SearchConfiguration.cs  7
- src/Products/Search/Entity/Web/Response/SummarySearchResult.cs  1
- src/Products/Search/Controllers/SearchApiController.cs  10
- src/Products/Search/Entity/Web/Response/SearchDocumentResult.cs  1
- src/Products/Search/Util/Directory/DirectoryUtils.cs  4
- src/Products/Search/Service/SearchService.cs  13
         

Clones added
============
- src/Products/Search/Config/SearchConfiguration.cs  1
- src/Products/Search/Controllers/SearchApiController.cs  7
- src/Products/Search/Util/Directory/DirectoryUtils.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5558557&bid=18230951)

File: src/Products/Search/Service/SearchService.cs [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+
+namespace GroupDocs.Total.MVC.Products.Search.Service
+{
+    public static class SearchService
+    {
+        private static Index index;
+
+        public static Dictionary<string, string> FileIndexingStatusDict { get; } = new Dictionary<string, string>();
+        public static Dictionary<string, string> PassRequiredStatusDict { get; } = new Dictionary<string, string>();
+        public static List<char> specialCharsList = new List<char>();
```
Comment (bobkovalex) [inline src/Products/Search/Service/SearchService.cs:22]: Codacy Issue found: [Make this field 'private' and encapsulate it in a 'public' property.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5558557)

File: src/Products/Search/Service/SearchService.cs [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+        public static List<char> foundSpecialChars = new List<char>();
```
Comment (bobkovalex) [inline src/Products/Search/Service/SearchService.cs:23]: Codacy Issue found: [Change the visibility of 'foundSpecialChars' or make it 'const' or 'readonly'.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5558557)

Comment (bobkovalex) [inline src/Products/Search/Service/SearchService.cs:23]: Codacy Issue found: [Make this field 'private' and encapsulate it in a 'public' property.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5558557)

**Comment by egorovpavel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/Products/Search/Entity/Web/Response/SearchDocumentResult.cs  1
- src/Products/Search/Util/Directory/DirectoryUtils.cs  4
- src/Products/Search/Controllers/SearchApiController.cs  10
- src/Products/Search/Entity/Web/Response/SummarySearchResult.cs  1
- src/Products/Search/Entity/Web/IndexedFileDescriptionEntity.cs  1
- src/Products/Search/Entity/Web/Request/SearchPostedData.cs  1
- src/Products/Search/Service/SearchService.cs  13
- src/Products/Search/Config/SearchConfiguration.cs  7
         

Clones added
============
- src/Products/Search/Util/Directory/DirectoryUtils.cs  1
- src/Products/Search/Controllers/SearchApiController.cs  7
- src/Products/Search/Config/SearchConfiguration.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GroupDocs/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5558558&bid=18230952)

**Comment by bobkovalex** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/Products/Search/Entity/Web/Request/SearchPostedData.cs  1
- src/Products/Search/Util/Directory/DirectoryUtils.cs  4
- src/Products/Search/Entity/Web/Response/SummarySearchResult.cs  1
- src/Products/Search/Service/SearchService.cs  13
- src/Products/Search/Config/SearchConfiguration.cs  7
- src/Products/Search/Controllers/SearchApiController.cs  10
- src/Products/Search/Entity/Web/Response/SearchDocumentResult.cs  1
- src/Products/Search/Entity/Web/IndexedFileDescriptionEntity.cs  1
         

Clones added
============
- src/Products/Search/Util/Directory/DirectoryUtils.cs  1
- src/Products/Search/Config/SearchConfiguration.cs  1
- src/Products/Search/Controllers/SearchApiController.cs  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5558557&bid=18230951)

**Comment by bobkovalex** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/Products/Search/Entity/Web/Response/SummarySearchResult.cs  1
- src/Products/Search/Controllers/SearchApiController.cs  10
- src/Products/Search/Util/Directory/DirectoryUtils.cs  4
- src/Products/Search/Entity/Web/Response/SearchDocumentResult.cs  1
- src/Products/Search/Entity/Web/Request/SearchPostedData.cs  1
- src/Products/Search/Config/SearchConfiguration.cs  7
- src/Products/Search/Service/SearchService.cs  13
- src/Products/Search/Entity/Web/IndexedFileDescriptionEntity.cs  1
         

Clones added
============
- src/Products/Search/Controllers/SearchApiController.cs  5
- src/Products/Search/Util/Directory/DirectoryUtils.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5558557&bid=18230951)

**Comment by egorovpavel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/Products/Search/Util/Directory/DirectoryUtils.cs  4
- src/Products/Search/Entity/Web/Response/SearchDocumentResult.cs  1
- src/Products/Search/Controllers/SearchApiController.cs  10
- src/Products/Search/Entity/Web/Request/SearchPostedData.cs  1
- src/Products/Search/Entity/Web/IndexedFileDescriptionEntity.cs  1
- src/Products/Search/Config/SearchConfiguration.cs  7
- src/Products/Search/Entity/Web/Response/SummarySearchResult.cs  1
- src/Products/Search/Service/SearchService.cs  13
         

Clones added
============
- src/Products/Search/Util/Directory/DirectoryUtils.cs  1
- src/Products/Search/Controllers/SearchApiController.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GroupDocs/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5558558&bid=18230952)

**Comment by egorovpavel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/client/apps/search-e2e/src/plugins/index.js  1
- src/Products/Search/Entity/Web/Response/SearchDocumentResult.cs  1
- GroupDocs.Total.MVC.Test/SearchControllerTest.cs  1
- src/Products/Search/Controllers/SearchApiController.cs  10
- src/Products/Search/Entity/Web/IndexedFileDescriptionEntity.cs  1
- src/Products/Search/Util/Directory/DirectoryUtils.cs  4
- src/Products/Search/Config/SearchConfiguration.cs  7
- src/client/apps/search-e2e/src/support/app.po.ts  1
- src/Products/Search/Entity/Web/Request/SearchPostedData.cs  1
- src/Products/Search/Service/SearchService.cs  13
- src/Controllers/SearchController.cs  1
- src/Products/Search/Entity/Web/Response/SummarySearchResult.cs  1
         

Clones added
============
- GroupDocs.Total.MVC.Test/SearchControllerTest.cs  6
- src/Products/Search/Controllers/SearchApiController.cs  5
- src/Products/Search/Util/Directory/DirectoryUtils.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GroupDocs/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5558558&bid=18230952)

**Comment by bobkovalex** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/client/apps/search-e2e/src/support/app.po.ts  1
- src/Products/Search/Service/SearchService.cs  13
- src/Products/Search/Entity/Web/Response/SearchDocumentResult.cs  1
- GroupDocs.Total.MVC.Test/SearchControllerTest.cs  1
- src/Products/Search/Entity/Web/Response/SummarySearchResult.cs  1
- src/Products/Search/Util/Directory/DirectoryUtils.cs  4
- src/Products/Search/Entity/Web/Request/SearchPostedData.cs  1
- src/Controllers/SearchController.cs  1
- src/Products/Search/Config/SearchConfiguration.cs  7
- src/Products/Search/Entity/Web/IndexedFileDescriptionEntity.cs  1
- src/Products/Search/Controllers/SearchApiController.cs  10
- src/client/apps/search-e2e/src/plugins/index.js  1
         

Clones added
============
- GroupDocs.Total.MVC.Test/SearchControllerTest.cs  6
- src/Products/Search/Util/Directory/DirectoryUtils.cs  1
- src/Products/Search/Controllers/SearchApiController.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5558557&bid=18230951)

**Comment by bobkovalex** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/Products/Search/Util/Directory/DirectoryUtils.cs  4
- GroupDocs.Total.MVC.Test/SearchControllerTest.cs  1
- src/Products/Search/Entity/Web/Response/SummarySearchResult.cs  1
- src/Products/Search/Service/SearchService.cs  13
- src/Controllers/SearchController.cs  1
- src/Products/Search/Entity/Web/Request/SearchPostedData.cs  1
- src/Products/Search/Config/SearchConfiguration.cs  7
- src/Products/Search/Entity/Web/IndexedFileDescriptionEntity.cs  1
- src/client/apps/search-e2e/src/plugins/index.js  1
- src/client/apps/search-e2e/src/support/app.po.ts  1
- src/Products/Search/Entity/Web/Response/SearchDocumentResult.cs  1
- src/Products/Search/Controllers/SearchApiController.cs  10
         

Clones added
============
- src/Products/Search/Util/Directory/DirectoryUtils.cs  2
- GroupDocs.Total.MVC.Test/SearchControllerTest.cs  6
- src/Products/Search/Controllers/SearchApiController.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5558557&bid=18230951)

**Comment by egorovpavel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- GroupDocs.Total.MVC.Test/SearchControllerTest.cs  1
- src/Products/Search/Config/SearchConfiguration.cs  7
- src/client/apps/search-e2e/src/support/app.po.ts  1
- src/Products/Search/Entity/Web/IndexedFileDescriptionEntity.cs  1
- src/Products/Search/Entity/Web/Request/SearchPostedData.cs  1
- src/Products/Search/Entity/Web/Response/SummarySearchResult.cs  1
- src/Products/Search/Controllers/SearchApiController.cs  10
- src/Controllers/SearchController.cs  1
- src/Products/Search/Util/Directory/DirectoryUtils.cs  4
- src/Products/Search/Entity/Web/Response/SearchDocumentResult.cs  1
- src/client/apps/search-e2e/src/plugins/index.js  1
- src/Products/Search/Service/SearchService.cs  13
         

Clones added
============
- GroupDocs.Total.MVC.Test/SearchControllerTest.cs  6
- src/Products/Search/Controllers/SearchApiController.cs  5
- src/Products/Search/Util/Directory/DirectoryUtils.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GroupDocs/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5558558&bid=18230952)

**Pull request merged**